cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/CapacityGoal.java

```diff
@@ -173,10 +173,13 @@ protected void initGoalState(ClusterModel clusterModel, OptimizationOptions opti
                         _balancingConstraint.capacityThreshold(resource())));
     }
     clusterModel.trackSortedReplicas(sortName(),
+                                     optimizationOptions.onlyMoveImmigrantReplicas() ? ReplicaSortFunctionFactory.selectImmigrants() :
+                                                                                       null,
```

Comment:
This change requires knowledge about what it means to set the value of the selection function to `null`, which can only be clear when the code is traced until `SortedReplicas#add(Replica)` -- i.e. note that the JavaDoc of `trackSortedReplicas(String, Function<Replica, Boolean>, Function<Replica, Integer>, Function<Replica, Double>)` does not specify what it means to pass `null`.

Can we either (1) keep the old way unless it is an RF increase, or (2) update the relevant JavaDoc for the expected behavior when `null` is passed?

Reply:
Update the Javadoc of `trackSortedReplicas(String, Function<Replica, Boolean>, Function<Replica, Integer>, Function<Replica, Double>)`